nixos/modules/system/activation/switch-to-configuration.py

```diff
@@ -0,0 +1,604 @@
+#! @python@
+
+import os.path
+import subprocess
+from syslog import openlog, syslog, LOG_NOTICE, LOG_ERR
+import sys
+
+out = "@out@"
+
+# To be robust against interruption, record what units need to be started etc.
+startListFile = "/run/systemd/start-list"
+restartListFile = "/run/systemd/restart-list"
+reloadListFile = "/run/systemd/reload-list"
+
+def die_with_usage():
+    print("""
+        Usage: {prog} [switch|boot|test]
+
+        switch:       make the configuration the boot default and activate now
+        boot:         make the configuration the boot default
+        test:         activate the configuration, but don\'t make it the boot default
+        dry-activate: show what would be done if this configuration were activated
+    """.format(prog=sys.argv[0]), file=sys.stderr)
+    sys.exit(1)
+
+def die(why):
+    print("error:", why, file=sys.stderr)
+    sys.exit(1)
+
+def read_if_exists(path, otherwise=None):
+    if os.path.isfile(path):
+        with open(path, 'r') as f:
+            return f.read()
+    else:
+        return otherwise
+
+if len(sys.argv) < 1:
+    die_with_usage()
+
+action = sys.argv[1]
```

Comment:
Can you add those lines in a one main function instead of mixing functions and code?